src/components/Dropdowns/SelectionDropdown.tsx

```diff
@@ -1,107 +1,123 @@
   selected: string;
-  /** An array of ItemType that describes the item properties (`{key, label}`),
+  /** An array of SelectItems that describes the item properties (`{value, label}`),
    * or an array of strings. Dropdown items will be presented in the provided array order.
    *
-   * If a string array is provided, ItemType objects will be
-   * auto-generated with `key` and `label` values set to the string.*/
-  items: ItemType[] | string[];
+   * If a string array is provided, SelectItems objects will be
+   * auto-generated with `value` and `label` values set to the string.*/
+  items: SelectItem[] | string[];
   disabled?: boolean;
   /** The type of button to render for the dropdown. See Antd's button types:
    * https://ant.design/components/button#components-button-demo-basic */
   buttonType?: ButtonProps["type"] | "outlined";
   /** Callback that is fired whenever an item in the dropdown is selected.
-   * The callback will be passed the `key` of the selected item. */
-  onChange: (key: string) => void;
-  showTooltip?: boolean;
+   * The callback will be passed the `value` of the selected item. */
+  onChange: (value: string) => void;
+  /** * If true, shows the label of the currently-selected item as a tooltip
+   * when hovering over the input/selection area. */
+  showSelectedItemTooltip?: boolean;
   /** Width of the dropdown. Overrides the default sizing behavior if set. */
   width?: string | null;
-  /**
-   * Whether the search bar should be enabled. If enabled, will show search bar and filter
-   * by search input when the total number of items is above `searchThresholdCount`. True by default.
-   */
-  enableSearch?: boolean;
-  /** The number of items that must be in the original list before the search bar will be shown. 10 by default.*/
-  searchThresholdCount?: number;
 };
 
 const defaultProps: Partial<SelectionDropdownProps> = {
-  label: null,
-  disabled: false,
-  buttonType: "outlined",
-  showTooltip: true,
-  width: null,
-  enableSearch: true,
-  searchThresholdCount: 10,
+  showSelectedItemTooltip: true,
+};
+
+// Override options in the menu list to include tooltips.
+const Option = (props: OptionProps): ReactElement => {
+  // Debounce the tooltip so it only shows after a short delay when focusing/hovering over it.
+  const isFocused = useDebounce(props.isFocused, 100) && props.isFocused;
+  const title = (props as OptionProps<SelectItem>).data.tooltip;
+  return (
+    <Tooltip
+      title={(props as OptionProps<SelectItem>).data.tooltip}
+      trigger={["hover", "focus"]}
+      placement="right"
+      open={title !== undefined && isFocused ? true : undefined}
+      mouseEnterDelay={0.5}
+      mouseLeaveDelay={0}
+    >
+      <div>
+        <components.Option {...props} />
+      </div>
+    </Tooltip>
+  );
 };
 
+/** Converts an array of strings or SelectItems to an array of SelectItems. */
+function formatAsSelectItems(items: string[] | SelectItem[]): SelectItem[] {
```

Comment:
I'm a little confused why this is necessary. Is this inconsistently in the datasets themselves? Could a change be made in your data processing to just always have this array bye of type "SelectItems" instead of doing a conversion here?

Reply:
Ah, this is for API convenience. I like that it lets components that use `SelectionDropdown` be unaware of the `SelectItem` type.